app/controllers/fhir/r4/api_controller.rb

```diff
@@ -744,7 +744,7 @@ def format_fhir_validation_errors(errors)
 
   def format_model_validation_errors(resource)
     resource.errors&.messages&.each_with_object([]) do |(attribute, errors), messages|
-      value = resource[attribute] || resource.public_send("#{attribute}_before_type_cast")
+      value = resource.public_send("#{attribute}_before_type_cast") || resource[attribute]
```

Comment:
What was the reasoning for this change? Just want to make sure I understand.

Reply:
This change was needed to deal with dates. If I send a patient with `birthDate: "22-11-1990` (`DD-MM-YYYY`), then Rails will parse that into November 22, 1990. But our validation will still reject that date, since we demand `YYYY-MM-DD` format. But if we use the value of the attribute _after_ typecast, it will be however Rails represents "November 22, 1990" as a string, which is `1990-11-22`.

So someone adds a Patient with a date in the `DD-MM-YYYY` format, but then gets an error saying something like:
```
"Validation Error for value '1990-11-22' on 'Date of Birth': is not a valid date, please use the 'YYYY-MM-DD' format"
```
which is kinda confusing, since the date being shown _is_ in the correct format, but that date isn't the one the user originally entered. This doesn't happen if we use the `_before_type_cast` version, since this will just be the user entered string.

Reply:
Gotcha, so given this is date specific should we make this a conditional check on attributes we expect to need this rather than on all attributes? I'm wondering if there could ever be unintended side-affects of calling `_before_type_cast` on other attributes aside from dates.

Reply:
Yeah that makes sense, I will make the `_before_type_cast` call apply explicitly to only dates.

Reply:
Done in https://github.com/SaraAlert/SaraAlert/pull/582/commits/3e908d1feeb71652836ad92d805b2b426b758ac1.